include/range/v3/numeric/partial_sum.hpp

```diff
@@ -50,19 +48,21 @@ namespace ranges
             tagged_pair<tag::in(I), tag::out(O)>
             operator()(I begin, S end, O result, S2 end_result, BOp bop = BOp{}, P proj = P{}) const
             {
-                using V = value_type_t<I>;
-                using X = concepts::Invocable::result_t<P&, V>;
-                coerce<V> v;
-                coerce<X> x;
-
+                using X = projected<projected<I, coerce<value_type_t<I>>>, P>;
+                using Y = reference_t<X>;
+                coerce<value_type_t<I>> v;
                 if(begin != end && result != end_result)
                 {
-                    auto t(x(invoke(proj, v(*begin))));
+                    auto &&tmp1 = *begin;
+                    auto &&tmp2 = v((decltype(tmp1) &&) tmp1);
+                    detail::decay_t<Y> t(invoke(proj, (decltype(tmp2) &&) tmp2));
```

Comment:
I think that coerce is at all necessary because we need a type for "accumulator" in the numeric algorithms. The goal is not to allow the proxy reference to be used as this type. So it seems that using `coerce` to `value_type` of the iterator should prevent this - if the projection does not itself contain some kind of UB, it has no access to the original proxy and thus cannot return it back to us. 
Then, decaying the result should give us a safe accumulator type (again, barring 'evil' types which have some weird side-effects). With this logic, the dance with storing references to extend some temporary's lifetimes (this is the original intent, is it not?) indeed seems unnecessary.